tasks/main.yml

```diff
@@ -1 +1,40 @@
 # role tasks
+
+- block:
+  - name: install NFS server packages
+    package:
+      name: "{{ item }}"
+      state: present
+    loop: "{{ nfs_server_packages }}"
+
+  - name: optionally create share directories
+    file:
+      path: "{{ item.share_path }}"
+      state: directory
+      owner: "{{ item.owner | default(omit) }}"
+      group: "{{ item.group | default(omit) }}"
+      mode: "{{ item.mode | default(omit) }}"
+    loop: "{{ nfs_server_shares }}"
+    when: item.create_dir
+
+  - name: add shares to {{ nfs_server_exports_file }}
+    template:
+      src: exports
+      dest: "{{ nfs_server_exports_file }}"
+      # It would be ideal to use the 'validate' option here, but the 'exportfs'
+      # command, while it can report errors, also takes action on any valid
+      # lines in the 'exports' file.  We don't want the 'validate' option to
+      # take any action, so we must depend on our handler to report syntax
+      # errors in the exports file to the user.
+      # validate: 'exportfs %s'
+    notify: export NFS shares
+    when: nfs_server_shares | length > 0
```

Comment:
Out of curiosity, does `when: nfs_server_shares` work here? I'd still expect empty iterables to evaluate as false in `when` clauses.

Reply:
It does.  I have no idea why I thought that was a good idea, and realize that it's not something you should normally have to do in Ansible.  :hankey: